frontend/public/components/graphs/area.tsx

```diff
@@ -130,9 +135,12 @@ export const AreaChart: React.FC<AreaChartProps> = ({
             theme={theme}
             scale={{ x: 'time', y: 'linear' }}
             padding={padding}
-            {...(allZero && { domain: { y: [0, 1] } })}
+            domain={{
+              ...(allZero && { y: [0, 1] }),
+              ...domain,
+            }}
           >
-            {xAxis && <ChartAxis tickCount={tickCount} tickFormat={formatDate} />}
+            {xAxis && <ChartAxis tickCount={tickCount} tickFormat={(tick) => formatDate(tick)} />}
```

Comment:
Using `(tick) => formatDate(tick)` instead of just `formatDate` might cause `ChartAxis` to re-render unnecessarily too. An easy solution to avoid that possibility would be to define `const xAxisTickFormat = (tick) => formatDate(tick);` outside the component.

Reply:
We can't define this outside the component since `formatDate` is a prop. I'll use the React callback hook to make sure the function definition only changes when necessary.